routers/user/auth.go

```
@@ -236,6 +243,90 @@ func SignUpPost(ctx *context.Context, cpt *captcha.Captcha, form auth.RegisterFo
 
 	ctx.Redirect(setting.AppSubUrl + "/user/login")
 }
+func OauthAuthorize(ctx *context.Context) {
+
+	ctx.Data["Title"] = ctx.Tr("oauth/authorize")
+	conf := &oauth2.Config{
+		ClientID:     setting.Cfg.Section("oauth").Key("CLIENTID").String(),
+		ClientSecret: setting.Cfg.Section("oauth").Key("CLIENTSECRET").String(),
+		RedirectURL:  setting.Cfg.Section("oauth").Key("REDIRECTURL").String(),
+		Scopes:       []string{setting.Cfg.Section("oauth").Key("SCOPE").String()},
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  setting.Cfg.Section("oauth").Key("AUTHURL").String(),
+			TokenURL: setting.Cfg.Section("oauth").Key("TOKENURL").String(),
+		},
+	}
+
+	url := conf.AuthCodeURL("state", oauth2.AccessTypeOnline)
+	ctx.Redirect(setting.AppSubUrl + url)
+}
+
+func OauthRedirect(ctx *context.Context) {
+	//request token using code
+	code := ctx.Query("code")
+	v := url.Values{}
+	v.Add("code", code)
+	v.Add("client_id", setting.Cfg.Section("oauth").Key("CLIENTID").String())
+	v.Add("client_secret", setting.Cfg.Section("oauth").Key("CLIENTSECRET").String())
+	v.Add("redirect_uri", setting.Cfg.Section("oauth").Key("REDIRECTURL").String())
+	v.Add("state", "state")
+	tokenResponse, _ := http.Post(setting.Cfg.Section("oauth").Key("TOKENURL").String(), "application/x-www-form-urlencoded", strings.NewReader(v.Encode()))
+
+	//parse tokenResponse
+	tokenbd, _ := ioutil.ReadAll(tokenResponse.Body)
```

Parsing data without validating if it succeeded (ignoring the error)?!

Now imagine that itsyou.online failed with an unexpected error. It returns an html document instead of the expected json. 
The error will go undetected because you ignored both the status code, and the parsing errors.